File: ArrayAdapterSpinner/app/src/main/java/com/example/arrayadapterspinner/MainActivity.java
package com.example.arrayadapterspinner;

import androidx.appcompat.app.AppCompatActivity;
import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Spinner;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    private Spinner chioceSpinner;
    private ListView listView;
    private TextView lblOutput;
    private ArrayAdapter<CharSequence> listAdapter;
    private String selectedMainCourse = "";
    private String selectedSideDish = "";
    private String selectedDrink = "";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        chioceSpinner = findViewById(R.id.chioce);
        listView = findViewById(R.id.listview);
        lblOutput = findViewById(R.id.lblOutput);

        chioceSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                switch (position) {
                    case 0:
                        listAdapter = ArrayAdapter.createFromResource(MainActivity.this,
                                R.array.Main_Course, android.R.layout.simple_list_item_1);
                        break;
                    case 1:
                        listAdapter = ArrayAdapter.createFromResource(MainActivity.this,
                                R.array.Side_Dish, android.R.layout.simple_list_item_1);
                        break;
                    case 2:
                        listAdapter = ArrayAdapter.createFromResource(MainActivity.this,
                                R.array.Drink, android.R.layout.simple_list_item_1);
                        break;
                }
                listView.setAdapter(listAdapter);
                listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
                    @Override
                    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                        String selectedItem = (String) parent.getItemAtPosition(position);
                        switch (chioceSpinner.getSelectedItemPosition()) {
                            case 0:
                                selectedMainCourse = selectedItem;
                                break;
                            case 1:
                                selectedSideDish = selectedItem;
                                break;
                            case 2:
                                selectedDrink = selectedItem;
                                break;
                        }
                        updateOutputLabel();
                    }
                });
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {

            }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        if (id == R.id.sent) {
            Intent intent = new Intent(MainActivity.this, DisplayActivity.class);
            intent.putExtra("output", lblOutput.getText().toString());
            startActivity(intent);
            return true;
        } else if (id == R.id.cannel) {
            selectedMainCourse = "";
            selectedSideDish = "";
            selectedDrink = "";
            lblOutput.setText("");
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    private void updateOutputLabel() {
        String output = "主餐: " + selectedMainCourse + "\n" +
                "副餐: " + selectedSideDish + "\n" +
                "飲料: " + selectedDrink;
        lblOutput.setText(output);
    }
}

File: 419homework/app/src/main/java/com/example/a419homework/MainActivity.kt
package com.example.a419homework

import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.View
import android.widget.EditText
import android.widget.RadioButton
import android.widget.RadioGroup
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity(), RadioGroup.OnCheckedChangeListener, TextWatcher {
    private var output: TextView? = null
    private var txt: EditText? = null
    private var rg: RadioGroup? = null
    private var type: RadioGroup? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Get view objects
        output = findViewById<View>(R.id.lblOutput) as TextView
        rg = findViewById<View>(R.id.rgGender) as RadioGroup
        txt = findViewById<View>(R.id.txtName) as EditText
        type = findViewById<View>(R.id.type) as RadioGroup

        // Register listener objects
        rg!!.setOnCheckedChangeListener(this)
        type!!.setOnCheckedChangeListener(this)
        txt!!.addTextChangedListener(this)
    }

    override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}

    override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {
        show(type!!.checkedRadioButtonId)
    }

    override fun afterTextChanged(editable: Editable) {}

    override fun onCheckedChanged(radioGroup: RadioGroup, i: Int) {
        show(i)
    }

    fun show(i: Int) {
        // 取得選擇的票價
        var ticketPrice = 0
        when (i) {
            R.id.FFT -> ticketPrice = 500
            R.id.ST -> ticketPrice = 400
            R.id.CT -> ticketPrice = 250
        }

        // 取得購買張數，如果輸入不是數字，則預設為 0
        val numberOfTickets = txt!!.text.toString().toIntOrNull() ?: 0

        // 計算總票價
        val totalPrice = numberOfTickets * ticketPrice

        // 取得性別文字
        val genderText = when (rg!!.checkedRadioButtonId) {
            R.id.rdbBoy -> "男生"
            R.id.rdbGirl -> "女生"
            else -> ""
        }

        // 取得票種文字
        val ticketTypeText = when (i) {
            R.id.FFT -> "全票"
            R.id.ST -> "學生票"
            R.id.CT -> "兒童票"
            else -> ""
        }

        // 更新 lblOutput 顯示總金額和其他信息
        output!!.text = "$genderText\n$ticketTypeText ${txt!!.text}\n金額$totalPrice 元"
    }
}
